### DISCUSSION TITLE
How can I clean the build cache?

### DISCUSSION THREAD

**Discussion by thomaszdxsn** [vercel/community#457] [Help] [answered]
Same [issue](https://github.com/zeit/next.js/issues/12820) in Next.js.

Resolved by delete node_modules and reinstall.

But dont work in vercel deployment, because it use the build cache. 

How to clean build cache?

**Comment by styfle** [comment, marked as answer]
Hi @thomaszdxsn 

If you're deploying using the git integration, you can click the "Redeploy" button.

- Visit the [Dashboard](https://vercel.com/dashboard)
- Click the deployment
- Click the three dots icon next to the "Visit" button
- Click "Redeploy"

If you deploy with Vercel CLI you can use `vercel --force` flag.

**Comment by ZeldOcarina** [comment]
Is there a way to not cache deployments in the first place? It keeps creating failed builds..

**Reply by dhi-cdretke** [reply]
Also having the same issue now for days, im blocked

**Reply by ZeldOcarina** [reply]
You can manually force a clean deployment, did you notice?

**Reply by koolamusic** [reply]
I noticed, I just wish there was a way I could manually clean that cache, so it generates a new one

**Comment by yactouat** [comment]
hey, jumping into the conversation => is there a way to clear the build cache using the REST API ?

**Reply by styfle** [reply]
Yes, add `?forceNew=1` 

https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment

**Reply by yactouat** [reply]
thanks ! this was the one place I didnt scroll to

**Comment by othecos** [comment]
Hey, you can also set this env variable `VERCEL_FORCE_NO_BUILD_CACHE=1`, as it says on vercel docs: 
[https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache)
